lib/std/os.zig

```diff
@@ -318,8 +327,8 @@ pub fn read(fd: fd_t, buf: []u8) ReadError!usize {
             EINTR => continue,
             EINVAL => unreachable,
             EFAULT => unreachable,
-            EAGAIN => if (std.event.Loop.instance) |loop| {
-                loop.waitUntilFdReadable(fd);
+            EAGAIN => if (is_async and std.event.Loop.instance != null) {
```

Comment:
Why this change? As far as I can tell nothing is different here. When that value is null it is comptime known to be null.

Reply:
I did it like that so even if there's an event loop it can still be forced to use a blocking version (to fix the thread startup using `extern`).

Reply:
If the file descriptor is known to be opened in blocking mode, then the best thing to do is noasync fn calls. This way the same (async) function is used, but bloat of an identical non async function is avoided, and it does not cause the caller to be async.

Reply:
Makes sense. So `noasync os.write()` would work even if os.write is async?

I more or less did this to see what impact it had on the server performance. It didn't seem to change much (which makes sense for high load since it's not io bound).